Sources/BraveWallet/Crypto/Stores/UserAssetsStore.swift

```diff
@@ -83,14 +83,34 @@ public class UserAssetsStore: ObservableObject {
       let visibleAssetIds = userAssets.filter(\.visible).map(\.id)
       blockchainRegistry.allTokens(network.chainId, coin: network.coin) { [self] registryTokens in
         allTokens = registryTokens + [network.nativeToken]
-        assetStores = allTokens.union(userAssets, f: { $0.id }).map { token in
-          AssetStore(
+        var uniqueAssets: [BraveWallet.BlockchainToken] = []
+        // we need to swap out the ERC721 assets that are from token registry but was added back as custom tokens with token id.
+        for registeryToken in allTokens {
+          if let matchedUserAsset = userAssets.first(where: { $0.id == registeryToken.id }) {
+            uniqueAssets.append(matchedUserAsset)
+          } else {
+            uniqueAssets.append(registeryToken)
+          }
+        }
+        
+        assetStores = uniqueAssets.union(userAssets, f: { $0.id }).map { token in
+          var isCustomToken: Bool {
+            if token.contractAddress.isEmpty {
+              return false
+            }
+            // Any token with a tokenId should be considered a custom token.
+            if !token.tokenId.isEmpty {
```

Comment:
Should this be specific to ERC721? In [EditUserAssetsView](https://github.com/brave/brave-ios/pull/6389/files#diff-54eee099ef64e253287bb50597d410abde1f4177ebe8382f111f8ed37a3bcbe4R20) we are using `isErc721` check as well, probably want to sync for that.

Reply:
no. its not only for ERC721. all NFT needs to have a token Id
1. if erc721 from token registry has not yet been added as custom nft, user should not able to swipe to delete
2. once its been added as a custom nft with a token id value, it can be deleted 
3. after the custom asset version is deleted, next time, user will need to enter a token id again if they want to set the erc721 as a visible asset.

Reply:
> all NFT needs to have a token Id

Solana NFTs don't have `tokenId` though. If a Solana NFT asset gets added to token registry, would this allow swipe to delete (but likely fail to delete)?